global-hooks/discovery/cluster_domain.go

```diff
@@ -99,17 +113,25 @@ func applyClusterDomainFromDNSPodFilter(obj *unstructured.Unstructured) (go_hook
 }
 
 func discoveryClusterDomain(input *go_hook.HookInput) error {
-	clusterDomainCoreCMSnap := input.Snapshots[clusterDomainCoreCMSnapName]
-	clusterDomainDNSPodsSnap := input.Snapshots[clusterDomainDNSPodsSnapName]
+	// We have a hook for handling clusterConfiguration.
+	// During the operation of this hook, there is a blocking check for filling in the clusterDomain field.
+	// So now we just need to check that the `clusterConfiguration` is present in the secrets.
+	// And if this is the case, then the `global.discovery.clusterDomain` will be filled.
+	currentConfig, ok := input.Snapshots["clusterConfiguration"]
+	if ok && len(currentConfig) > 0 {
+		return nil
```

Comment:
add a comment please, that the `cluster_configuration.go` hook handles this situation

Reply:
We already have comment:
```plain
	// We have a hook for handling clusterConfiguration.
	// During the operation of this hook, there is a blocking check for filling in the clusterDomain field.
	// So now we just need to check that the `clusterConfiguration` is present in the secrets.
	// And if this is the case, then the `global.discovery.clusterDomain` will be filled.
```